### PR TITLE
DEV/MNT: migrate to latest standards with pyproject.toml [LCLSPC-603]

### PR DESCRIPTION
**Pull request by klauer**
## Description
* Upgrading by way of https://github.com/pcdshub/pcds-python-migration-tools
* pcdsutils has been migrated to use setuptools-scm, replacing versioneer, as its version-string management tool of choice.
* pcdsutils has been migrated to use the modern ``pyproject.toml``, replacing ``setup.py`` and related files.
* Older language features and syntax found in the repository have been updated to Python 3.9+ standards by way of ``pyupgrade``.
* Sphinx 6.0 is now supported for documentation building. ``docs-versions-menu`` replaces ``doctr-versions-menu`` and ``doctr`` usage for documentation deployment on GitHub Actions.  The deployment key is now no longer required.
* `pyupgrade` was used to update annotations, though there might not have been many here
* `pycln` was used to clean up imports

## Motivation and Context
* Migrating because we want to migrate our projects from Travis CI for one

## How Has This Been Tested?
* GitHub Actions

## Where Has This Been Documented?
This PR and Jira [LCLSPC-603](https://jira.slac.stanford.edu/browse/LCLSPC-603)

### CODE REVIEW

**Review by ZLLentz — APPROVED** [review]
Looks normal to me, note that this repo doesn't actually have any real automated tests.

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
 lcls-krtc
-six
```
Comment (tangkong) [inline requirements.txt:2]: This must be nice to remove

**Review by tangkong — APPROVED** [review]
Looks good to me.  The entrypoints got moved and everything looks normal.

File: requirements.txt (hunk 1 above)
Comment (klauer) [reply requirements.txt:2]: Yeah, a whole dependency for a single standard library function...

**Pull request merged**